src/app/clusters/commissioner-control-server/commissioner-control-server.cpp

```diff
@@ -0,0 +1,251 @@
+    auto requestId    = commandData.requestId;
+    auto vendorId     = commandData.vendorId;
+    auto productId    = commandData.productId;
+    auto & label      = commandData.label;
+
+    // Create a CommissioningApprovalRequest struct and populate it with the command data
+    Clusters::CommissionerControl::CommissioningApprovalRequest request = { .requestId    = requestId,
+                                                                            .vendorId     = vendorId,
+                                                                            .productId    = productId,
+                                                                            .clientNodeId = sourceNodeId,
+                                                                            .fabricIndex  = fabricIndex,
+                                                                            .label        = label };
+
+    Clusters::CommissionerControl::Delegate * delegate =
+        Clusters::CommissionerControl::CommissionerControlServer::Instance().GetDelegate();
+
+    VerifyOrExit(sourceNodeId != kUndefinedNodeId, err = CHIP_ERROR_WRONG_NODE_ID);
+    VerifyOrExit(delegate != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+
+    // Handle commissioning approval request
+    err = delegate->HandleCommissioningApprovalRequest(request);
+
+exit:
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "emberAfCommissionerControlClusterRequestCommissioningApprovalCallback error: %s", err.AsString());
+        status = Status::Failure;
+    }
+
+    commandObj->AddStatus(commandPath, status);
+    return true;
+}
+
+bool emberAfCommissionerControlClusterCommissionNodeCallback(
+    app::CommandHandler * commandObj, const app::ConcreteCommandPath & commandPath,
+    const Clusters::CommissionerControl::Commands::CommissionNode::DecodableType & commandData)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    Status status  = Status::Success;
+
+    ChipLogProgress(Zcl, "Received command to commission node");
+
+    auto sourceNodeId = getNodeId(commandObj);
+    auto requestId    = commandData.requestId;
+
+    auto info       = std::make_unique<Clusters::CommissionerControl::CommissionNodeInfo>();
+    info->ipAddress = commandData.ipAddress;
+    info->port      = commandData.port;
+
+    Clusters::CommissionerControl::Delegate * delegate =
+        Clusters::CommissionerControl::CommissionerControlServer::Instance().GetDelegate();
+
+    VerifyOrExit(sourceNodeId != kUndefinedNodeId, err = CHIP_ERROR_WRONG_NODE_ID);
+    VerifyOrExit(delegate != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+
+    // Handle commissioning approval request
+    err = delegate->ValidateCommissionNodeCommand(sourceNodeId, requestId, info->params);
+    VerifyOrExit(err == CHIP_NO_ERROR, err = CHIP_ERROR_INCORRECT_STATE);
+
+    // Add the response for the commissioning window
+    AddReverseOpenCommissioningWindowResponse(commandObj, commandPath, info->params);
```

Comment:
Is the result ready at this point. I don't think it is you are sending a response before you have the result because you are scheduling the work. I think you need to do to respond to this asyncronously

Reply:
The client only send the  CommissionNode command after it received CommissioningRequestResult event with StatusCode of SUCCESS
  
```
    // Add the response for the commissioning window
    AddReverseOpenCommissioningWindowResponse(commandObj, commandPath, info->params);
```
    
This adds the ReverseOpenCommissioningWindow response synchronously, but the bridge won't send this response until return from the calling function.

To make sure the bridge send ReverseCommissionNode after ReverseOpenCommissioningWindow response, we need to use ScheduleWork to schedule this task after this function returns.